ocs_ci/utility/utils.py

```diff
@@ -626,8 +626,27 @@ def exec_cmd(
         cmd = shlex.split(cmd)
     if cluster_config and cmd[0] == "oc" and "--kubeconfig" not in cmd:
         kubepath = cluster_config.RUN["kubeconfig"]
-        cmd = list_insert_at_position(cmd, 1, ["--kubeconfig"])
-        cmd = list_insert_at_position(cmd, 2, [kubepath])
+        kube_index = 1
+        # check if we have an oc plugin in the command
+        plugin_list = "oc plugin list"
+        cp = subprocess.run(
+            shlex.split(plugin_list),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        subcmd = cmd[1].split("-")
+        if len(subcmd) > 1:
+            subcmd = "_".join(subcmd)
+        log.debug(f"searching for plugin: {subcmd}")
+
+        for l in cp.stdout.decode().splitlines():
+            if subcmd in l:
+                # If oc cmdline has plugin name then we need to push the
+                # --kubeconfig to next index
+                kube_index = 2
+        cmd = list_insert_at_position(cmd, kube_index, ["--kubeconfig"])
+        cmd = list_insert_at_position(cmd, kube_index + 1, [kubepath])
+        log.info(f"Modified CMD = {cmd}")
```

Comment:
Hmm, I realized that you are printing here the regular cmd instead of masked_cmd so whatever secret can be exposed by this logging message.
You will need to update the masked_cmd like we do in line 623 and then print the masked_cmd instead of just cmd here ;)

Reply:
ack, I had it for debugging purpose, I will remote it.

Reply:
done